### PR TITLE
Perception:update pointpillars network and models

### PR DESCRIPTION
**Pull request by panfengsu**

### CODE REVIEW

File: modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc [hunk 1]
```diff
@@ -403,8 +392,11 @@ void PointPillars::ConvertAnchors2BoxAnchors(float* anchors_px,
                                              float* box_anchors_max_x_,
                                              float* box_anchors_max_y_) {
   // flipping box's dimension
-  float flipped_anchors_dx[kNumAnchor] = {};
-  float flipped_anchors_dy[kNumAnchor] = {};
+  // float flipped_anchors_dx[kNumAnchor] = {};
+  // float flipped_anchors_dy[kNumAnchor] = {};
```
Comment (jeroldchen) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc:396]: Better just delete these lines.

File: modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc [hunk 2]
```diff
@@ -606,39 +625,64 @@ void PointPillars::DoInference(const float* in_points_array,
...
-      reinterpret_cast<float*>(rpn_buffers_[1]),
-      reinterpret_cast<float*>(rpn_buffers_[2]),
-      reinterpret_cast<float*>(rpn_buffers_[3]), dev_anchor_mask_,
-      dev_anchors_px_, dev_anchors_py_, dev_anchors_pz_, dev_anchors_dx_,
-      dev_anchors_dy_, dev_anchors_dz_, dev_anchors_ro_, dev_filtered_box_,
-      dev_filtered_score_, dev_filtered_label_, dev_filtered_dir_,
-      dev_box_for_nms_, dev_filter_count_, out_detections, out_labels);
+      reinterpret_cast<float*>(bbox_pred.data_ptr<float>()),
+      reinterpret_cast<float*>(cls_score.data_ptr<float>()),
+      reinterpret_cast<float*>(dir_cls_preds.data_ptr<float>()),
```
Comment (jeroldchen) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc:680]: As `data_ptr<float>()` is already `float*` type, there is no need for `reinterpret_cast<float*>` .

File: modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_test.cc [hunk 3]
```diff
@@ -37,14 +37,16 @@
  * @date 2019/02/26
  */
 
+#include "modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.h"
```
Comment (jeroldchen) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_test.cc:40]: This header is not its corresponding one. Just put it back to the right place.

File: modules/perception/production/conf/perception/perception_common.flag [hunk 4]
```diff
@@ -109,7 +109,7 @@
 # not be detected if you use 128c Lidar.
 # type: bool
 # default: true
```
Comment (jeroldchen) [inline modules/perception/production/conf/perception/perception_common.flag:111]: Looks like the default value is false now. Correct this commented line.

File: modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc [hunk 5]
```diff
@@ -99,15 +98,19 @@ const std::vector<std::vector<float>> PointPillars::kAnchorRo =
 PointPillars::PointPillars(const bool reproduce_result_mode,
                            const float score_threshold,
                            const float nms_overlap_threshold,
-                           const std::string pfe_onnx_file,
-                           const std::string rpn_onnx_file)
+                           const std::string pfe_torch_file,
+                           const std::string scattered_torch_file,
```
Comment (storypku) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc:102]: const std::string&

File: modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc [hunk 6]
```diff
@@ -99,15 +98,19 @@ const std::vector<std::vector<float>> PointPillars::kAnchorRo =
...
+                           const std::string backbone_torch_file,
```
Comment (storypku) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc:103]: const std::string&

File: modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.h [hunk 7]
```diff
@@ -325,14 +341,20 @@ class PointPillars {
...
    * @details Variables could be changed through point_pillars_detection
    */
   PointPillars(const bool reproduce_result_mode, const float score_threshold,
                const float nms_overlap_threshold,
-               const std::string pfe_onnx_file,
-               const std::string rpn_onnx_file);
+               const std::string pfe_torch_file,
+               const std::string scattered_torch_file,
+               const std::string backbone_torch_file,
+               const std::string fpn_torch_file,
```
Comment (storypku) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.h:356]: const std::string&

**Review by storypku — CHANGES_REQUESTED** [review]

**Pull request closed without merging**